app/components/header.tsx

```diff
@@ -1,11 +1,77 @@
-import { Text, Center } from '@chakra-ui/react';
+import {
+  Text,
+  Heading,
+  Avatar,
+  Flex,
+  HStack,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  IconButton,
+} from '@chakra-ui/react';
+import { TriangleUpIcon, LockIcon } from '@chakra-ui/icons';
+import { Link, useFetcher } from '@remix-run/react';
+
+import { useUser } from '~/utils';
 
 export default function Header() {
+  const user = useUser();
+  const fetcher = useFetcher();
   return (
     <header>
-      <Center>
-        <Text fontSize="2xl">Hi! I am the header</Text>
-      </Center>
+      <Flex
+        backgroundColor="brand.500"
+        height="70px"
+        alignItems="center"
+        shadow="2xl"
+        fontSize="18"
+      >
+        <HStack width="50%" color="white" justifyContent="flex-start" gap="10" marginLeft="10">
+          <Link to={{ pathname: '/domains' }}>
+            <Flex alignItems="center">
+              <TriangleUpIcon marginRight="2" />
+              <Text>Domains</Text>
+            </Flex>
+          </Link>
+
+          <Link to={{ pathname: '/certificate' }}>
+            <Flex alignItems="center">
+              <LockIcon marginRight="2" />
+              <Text>Certificate</Text>
+            </Flex>
+          </Link>
+        </HStack>
+        <Link to={{ pathname: '/' }}>
+          <Heading as="h1" size="xl" color="white">
+            My.Custom.Domain
+          </Heading>
+        </Link>
+        <Flex
+          width="50%"
+          justifyContent="flex-end"
+          alignItems="center"
+          color="white"
+          gap="5"
+          marginRight="10"
+        >
+          <Text>{user.username}</Text>
+          <Menu>
+            <MenuButton as={IconButton} icon={<Avatar size="sm" />} />
+            <MenuList color="black">
+              <MenuItem
+                onClick={() => {
```

Comment:
since it is one line function, you could just do 
```typescript
() => fetcher.submit({}, { method: 'post', action: '/logout' })
```
Not need for `{}`